numpy/core/src/multiarray/descriptor.c

```diff
@@ -1423,10 +1387,20 @@ PyArray_DescrConverter(PyObject *obj, PyArray_Descr **at)
             check_num = NPY_VOID;
         }
         else {
-            *at = _arraydescr_fromobj(obj);
+            *at = _arraydescr_from_dtype_attr(obj);
             if (*at) {
                 return NPY_SUCCEED;
             }
+
+            /*
+             * Note: this comes after _arraydescr_from_dtype_attr because the ctypes
+             * type might override the dtype if numpy does not otherwise
+             * support it.
+             */
+            if (npy_ctypes_check((PyTypeObject *)obj)) {
```

Comment:
Generally prefer to check against value, especially as the error return is '0'.

Reply:
Or maybe `NPY_SUCCEED` as that seems to be the interface for this bunch of functions.

Reply:
I'd rather leave this as is, since it matches things like `PyString_Check`